aggregateur/validators.py

```diff
@@ -342,3 +342,75 @@ def schema_json_data(self):
         with open(self.filepath(self.SCHEMA_FILENAME)) as f:
             self.schema_data = json.load(f)
         return self.schema_data
+
+
+class GenericValidator(BaseValidator):
+    SCHEMA_FILENAME = "schema.yml"
+
+    def __init__(self, repo):
+        super().__init__(repo)
+        self.schema_data = None
+        self.has_changelog = False
+
+    def schemas_metadata(self):
+        return [
+            {
+                "path": self.SCHEMA_FILENAME,
+                "original_path": self.SCHEMA_FILENAME,
+                "title": self.title,
+                "latest_url": self.schema_url(self.SCHEMA_FILENAME),
+            }
+        ]
+
+    @property
+    def title(self):
+        return self.get_schema_data()["title"]
+
+    @property
+    def description(self):
+        return self.get_schema_data()["description"]
+
+    @property
+    def homepage(self):
+        return self.get_schema_data()["homepage"]
+
+    def validate(self):
+        super().validate()
+        self.check_file_exists(self.SCHEMA_FILENAME)
+        self.check_schema(self.SCHEMA_FILENAME)
+        self.check_extra_keys()
+
+    def extract(self):
+        files = {
+            self.SCHEMA_FILENAME: self.filepath_or_none(self.SCHEMA_FILENAME),
+            "README.md": self.filepath_or_none("README.md"),
+            "SEE_ALSO.md": self.filepath_or_none("SEE_ALSO.md"),
+            "CONTEXT.md": self.filepath_or_none("CONTEXT.md"),
+        }
+
+        if self.is_latest_version():
+            files[self.CHANGELOG_FILENAME] = self.filepath_or_none(
+                self.CHANGELOG_FILENAME
+            )
```

Comment:
I think you may need to set `has_changelog`to `True` here. It's not pretty, feel feel to refactor other classes if you feel like it.

Reply:
Why? Won't it work just like on TableSchemaValidator (eg use a CHANGELOG.md if there's one)?

Reply:
(changed for the XSD pattern)